Src/Public/Invoke-AsBuiltReport.Rubrik.CDM.ps1

```diff
@@ -1089,12 +1089,18 @@ function Invoke-AsBuiltReport.Rubrik.CDM {
                             @{N = "SID"; E = { $_.sid } }, @{N = "SLA Domain"; E = { $_.effectiveSlaDomainName } },
                             @{N = "Assignment Type"; E = { $_.slaAssignment } }
                             Write-Verbose -Message "[Rubrik] [$($brik)] [Protected Objects] Retrieving Protected Filesets "
-                            $Filesets = Get-RubrikFileset -Relic:$false -PrimaryClusterID local -DetailedObject | Where-Object { $_.effectiveSlaDomainId -ne 'Unprotected' -and $null -eq $_.shareId } | Select-Object -Property @{N = "Hostname"; E = { $_.hostname } },
-                            @{N = "Operating System"; E = { $_.operatingSystemType } }, @{N = "Fileset Name"; E = { $_.name } },
-                            @{N = "SLA Domain"; E = { $_.effectiveSlaDomainName } }
+                            $Filesets = Get-RubrikFileset -Relic:$false -PrimaryClusterID local | Where-Object { $_.effectiveSlaDomainId -ne 'Unprotected' -and $null -eq $_.shareId }
+                            if ($null -ne $Filesets -and 0 -ne $Filesets[0].total ) {
+                                $Filesets = Get-RubrikFileset -Relic:$false -PrimaryClusterID local -DetailedObject | Where-Object { $_.effectiveSlaDomainId -ne 'Unprotected' -and $null -eq $_.shareId } | Select-Object -Property @{N = "Hostname"; E = { $_.hostname } },
+                                @{N = "Operating System"; E = { $_.operatingSystemType } }, @{N = "Fileset Name"; E = { $_.name } },
+                                @{N = "SLA Domain"; E = { $_.effectiveSlaDomainName } }
```

Comment:
Look to move away from using expressions in favour of `PSCustomPObjects` in alignment with the contributing [guidelines](https://github.com/AsBuiltReport/AsBuiltReport.Core/blob/dev/CONTRIBUTING.md#use-a-consistent-coding-style)

Reply:
Will tackle this in another PR as it's probably a big change!  I'll create an issue for it!